.github/workflows/apiBreakTest.yml

```diff
@@ -0,0 +1,216 @@
+name: Public Interface Breakage Detection
+
+on:
+  pull_request:
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  buildAndCheckAPIBreakage:
+    name: Build and Check API Breakage
+    runs-on: macos-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 #v4.1.1
+      with:
+        fetch-depth: 0
```

Comment:
Same as above: do we really need to fetch all history for all branches and tags?